### PR TITLE
Added Caravel merge and precheck in action + Updated SOFA-HD and QLSOFA-HD gds

### PR DESCRIPTION
**Pull request by ganeshgore**
+ added action performs merge with caravel, performs precheck and push to dummy repository
+ Fixes the reset problem in QLSOFA-HD [Tested]

TODO:
+ actions scripts needs to be extended for other designs that will be (Incremental process)

### CODE REVIEW

File: .github/workflows/quick_test.sh [hunk 1]
```diff
@@ -9,28 +9,24 @@ set -e
...
+# Post processing netlist to use custom cells
 python3 HDL/common/custom_cell_mux_primitive_generator.py --template_netlist HDL/k4_N8_reset_softadder_caravel_io_FPGA_12x12_customhd_cc/SRC/sub_module/mux_primitives.v --output_verilog HDL/k4_N8_reset_softadder_caravel_io_FPGA_12x12_customhd_cc/SRC/sub_module/mux_primitives_hd.v
 
 ##############################################
 # Generate wrapper HDL codes to bridge Caravel I/Os and FPGA I/Os
 python3 HDL/common/wrapper_lines_generator.py --template_netlist HDL/common/caravel_fpga_wrapper_hd_template.v --pin_assignment_file HDL/common/caravel_wrapper_pin_assignment_v1.0.json --output_verilog HDL/common/caravel_fpga_wrapper_hd_v1.0.v
 python3 HDL/common/wrapper_lines_generator.py --template_netlist HDL/common/caravel_fpga_wrapper_hd_template.v --pin_assignment_file HDL/common/caravel_wrapper_pin_assignment_v1.1.json --output_verilog HDL/common/caravel_fpga_wrapper_hd_v1.1.v
 
+##############################################
+# Generate wrapper HDL codes to bridge Caravel I/Os and FPGA I/Os
```
Comment (tangxifan) [inline .github/workflows/quick_test.sh:24]: It seems that there are some conflicts on this file, please check out the latest version.

File: SynRepoConfig/sync_files.csv [hunk 2]
```diff
@@ -0,0 +1,2 @@
+SrcLoc, DestLoc
```
Comment (tangxifan) [inline SynRepoConfig/sync_files.csv:1]: Can you briefly explain to me what is this file for?

Comment (ganeshgore) [reply SynRepoConfig/sync_files.csv:1]: This is currently a placeholder to copy files from SOFA to Caravel-SOFA-HD repo. 
like line 2 copies `FPGA1212_FLAT_HD_SKY_task` to `Caravel-SOFA-HD` with `OpenFPGA_task` name while deployment.

File: .github/workflows/quick_test.sh (hunk 1 above)
Comment (ganeshgore) [reply .github/workflows/quick_test.sh:24]: I should reset this file, I merged with old master.

**Review by tangxifan — APPROVED** [review]

**Pull request merged**